optional/src/main/java/org/dmfs/optional/single/Presence.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.optional.single;
+
+import org.dmfs.jems.single.Single;
+import org.dmfs.optional.Optional;
+
+
+/**
+ * A {@link Single} of the presence value of an {@link Optional}.
+ *
+ * @author Marten Gajda
+ */
+public final class Presence implements Single<Boolean>
```

Comment:
The use case I had in mind was to provide it as the condition to a `Conditional`, so you can create an `Optional` based on the presence of another `Optional`. Hmm … I just noticed that this could also be achieved with a `Mapping` decorator and a constant `Function`.
Anyway, I'll remove it from the commit.